rotkehlchen/globaldb/handler.py

```diff
@@ -487,51 +485,47 @@ def get_assets_mappings(identifiers: list[str]) -> tuple[dict[str, dict], dict[s
         return result, asset_collections
 
     @staticmethod
-    def search_assets(
-            filter_query: 'AssetsFilterQuery',
-            db: 'DBHandler',
-    ) -> list[dict[str, Any]]:
+    def search_assets(filter_query: 'AssetsFilterQuery', db: 'DBHandler') -> list[dict[str, Any]]:
         """Returns a list of asset details that match the search query provided."""
         search_result = []
-        globaldb = GlobalDBHandler()
-        query, bindings = filter_query.prepare()
-        query = ALL_ASSETS_TABLES_QUERY.format(dbprefix='globaldb.') + query
-        resolved_eth = A_ETH.resolve_to_crypto_asset()
-        globaldb._wal_checkpoint()
+        ignored_asset_handler = filter_query.ignored_assets_handling.get_handler()
```

Comment:
this is hard to read because you later do a `not ignored_asset_handler` that makes it complex to understand. Can you maybe find a better name to it and maybe change the logic so you don't have to do a `not logic`?